apps/dashboard/src/@/components/ui/CopyAddressButton.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { CheckIcon, CopyIcon } from "lucide-react";
+import { useState } from "react";
+import { cn } from "../../lib/utils";
+import { Button } from "./button";
+import { ToolTipLabel } from "./tooltip";
+
+export function CopyAddressButton(props: {
+  address: string;
+  className?: string;
+  iconClassName?: string;
+  variant?:
+    | "primary"
+    | "default"
+    | "destructive"
+    | "outline"
+    | "secondary"
+    | "ghost";
+}) {
+  const [isCopied, setIsCopied] = useState(false);
+  const shortenedAddress = `${props.address.slice(0, 6)}...${props.address.slice(0, 4)}`;
+
+  return (
+    <ToolTipLabel label="Copy address">
+      <Button
+        variant={props.variant || "outline"}
+        aria-label="Copy"
+        className={cn(
+          "h-auto w-auto px-2.5 py-1.5 flex gap-2 text-xs rounded-lg font-mono text-foreground",
+          props.className,
+        )}
+        onClick={() => {
+          navigator.clipboard.writeText(props.address);
```

Comment:
does this work cross browser? I remember there used to be a bunch of hacks we had to do for copy to clipboard before

Reply:
Browser support looks good https://caniuse.com/mdn-css_properties_aspect-ratio